test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -332,3 +497,154 @@ func checkData(t *testing.T, start time.Time, lineCount int) {
 	)
 	assert.NoError(t, err)
 }
+
+func ValidateEntity(t *testing.T, logGroup, logStream string, begin, end *time.Time, expectedEntity expectedEntity) {
```

Comment:
nonblocking but recommended: This function has a lot of overlaps with this function: https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/test/entity/entity_test.go#L241

Maybe good idea to move `ValidateEntity` to a common package and just use for any entity validation tests in the future.